### ISSUE TITLE
User Icons

### ISSUE THREAD

**Issue by elwinmartin** [LangCorrect/server#460]
This user suggested this in one of their English writing submissions but it really should also be opened as a potential improvement.

https://langcorrect.com/journals/need-thumbnail/

I'm not sure how file storage is currently handled for user uploaded data but a single icon per user at the resolution needed for browser would even fit in a DB as a BLOB datatype (not that this is a good idea). 

Implemented as normal files, this kind of item should be cached by a CDN presumably so the bottom line cost will not be very large, but I don't have a good estimate since I don't know the size of the user base. I think this has a pretty large impact on the feeling the site would have; it could probably be mocked pretty quickly as an attachment.

**Comment by danielzeljko** [comment]
We've had some discussions about this on our discord server in the past, and it's honestly a bit of a dilemma for me. On one hand, it would definitely spice up the platform making it more personal and engaging, but I worry it might steer focus away from learning. I've seen other educational apps turn into dating platforms, and that's not what I want for LC.

Still, I'm not completely against the idea. Maybe we can give it a test run, but I'd like to be crystal clear that selfies aren't allowed. Some other ideas that popped up in the discussions was using customizable avatars (like Reddit) or making profile pictures a premium feature with blurred images for standard users, Finding the right balance between personalization and keeping the focus on learning is sadly much easier said than done. I'd love to hear your thoughts!

We currently use local file storage (stored on the server), which has been a decent approach so far. I don't expect this to change anytime soon, but if the traffic picks up then we should definitely switch to something like S3 with CloudFront. Storing the pictures to the DB was an interesting idea and it led me to this detailed SO answer which you might find interesting: https://stackoverflow.com/questions/8922056/what-is-the-best-way-to-upload-and-store-pictures-on-the-site/8922090#8922090. I sure did!

**Comment by keyxcode** [comment]
I strongly agree with @danielzeljko here. Especially with the messenger feature being close to release, I think it's even more important to be careful and avoid turning LC into a casual "language dating" app.

Personally, I like the avatar builder approach of Reddit and Duolingo."

**Comment by keyxcode** [comment]
I strongly agree with @danielzeljko here. Especially with the messenger feature being close to release, I think it's even more important to be careful and avoid turning LC into a casual "language dating" app.

Personally, I like the avatar builder approach of Reddit and Duolingo.

**Comment by elwinmartin** [comment]
Yeah that's a very good point if the trust of the people using it is at risk then I think photo icons would absolutely be a detriment to the platform. I think personalization is good but not at the expense of harassment.  @danielzeljko

**Comment by keyxcode** [comment]
Btw I do think that some level of personalization is a great idea too! It's just that when apps overdo it, they end up attracting the wrong crowd, which is a shame :/ So it's probably better to err on the side of caution.
